src/linux_ip_socket/linux_ip_socket.h

```diff
@@ -0,0 +1,180 @@
+/**@file
+ * This file is part of the TASTE Linux Runtime.
+ *
+ * @copyright 2021 N7 Space Sp. z o.o.
+ *
+ * TASTE Linux Runtime was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LINUX_IP_SOCKET_H
+#define LINUX_IP_SOCKET_H
+
+/**
+ * @file     linux_ip_socket.h
+ * @brief    Driver for TASTE with uses TCP/IP for communication.
+ *
+ */
+
+#include <cstddef>
+#include <cstdint>
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <poll.h>
+
+#include <Thread.h>
+#include <system_spec.h>
+
+#include <drivers_config.h>
+
+extern "C"
+{
+#include <Broker.h>
+}
+
+/**
+ * @brief Structure for driver internal data.
+ *
+ * This structure is allocated by runtime and the pointer is passed to all driver functions.
+ * The name of this structure shall match driver definition from ocarina_components.aadl
+ * and has suffix '_private_data'.
+ */
+class linux_ip_socket_private_data final
+{
+  public:
+    /**
+     * @brief  Default constructor.
```

Comment:
`Default` here is a little bit misleading, as this sets some members to some values

Reply:
Changed.